internal/rpc/serializer.go

```diff
@@ -42,13 +39,22 @@ func SerializeFullBlocks(chainId *big.Int, blocks []RPCFetchBatchResult[common.R
 
 		result.Data.Block = serializeBlock(chainId, rawBlock.Result)
 		blockTimestamp := result.Data.Block.Timestamp
-		result.Data.Transactions = serializeTransactions(chainId, rawBlock.Result["transactions"].([]interface{}), blockTimestamp)
 
-		if rawLogs, exists := rawLogsMap[rawBlock.BlockNumber.String()]; exists {
-			if rawLogs.Error != nil {
-				result.Error = rawLogs.Error
+		if rawReceipts, exists := rawReceiptsMap[rawBlock.BlockNumber.String()]; exists {
+			if rawReceipts.Error != nil {
+				result.Error = rawReceipts.Error
 			} else {
-				result.Data.Logs = serializeLogs(chainId, rawLogs.Result, result.Data.Block)
+				result.Data.Logs = serializeLogsFromReceipts(chainId, &rawReceipts.Result, result.Data.Block)
+				result.Data.Transactions = serializeTransactions(chainId, rawBlock.Result["transactions"].([]interface{}), blockTimestamp, &rawReceipts.Result)
+			}
+		} else {
+			if rawLogs, exists := rawLogsMap[rawBlock.BlockNumber.String()]; exists {
+				if rawLogs.Error != nil {
+					result.Error = rawLogs.Error
+				} else {
+					result.Data.Logs = serializeLogs(chainId, rawLogs.Result, result.Data.Block)
+					result.Data.Transactions = serializeTransactions(chainId, rawBlock.Result["transactions"].([]interface{}), blockTimestamp, nil)
```

Comment:
we didn't serialize TXs before?

Reply:
We did, but now we need to add data from receipts if they exist